1st chunk:
disable libinput support otherwise module-mmkbd-evdev.so is built
3nd chunk:
Should match b_lundef in meson(1).

Index: src/modules/meson.build
--- src/modules/meson.build.orig
+++ src/modules/meson.build
@@ -80,12 +80,6 @@ endif
 
 # Modules enabled by headers
 
-if cc.has_header('linux/input.h')
-  all_modules += [
-    [ 'module-mmkbd-evdev', 'module-mmkbd-evdev.c' ],
-  ]
-endif
-
 if cdata.has('HAVE_OSS_OUTPUT')
   subdir('oss')
   all_modules += [
@@ -100,6 +94,14 @@ if cc.has_function('mkfifo')
   ]
 endif
 
+# sndio(7)
+if host_machine.system() == 'openbsd'
+  libsndio_dep = dependency('sndio', required: true)
+  all_modules += [
+    [ 'module-sndio', 'module-sndio.c', [], [], [libsndio_dep], [] ],
+  ]
+endif
+
 # Modules enabled by dependencies
 
 if alsa_dep.found()
@@ -310,7 +312,7 @@ foreach m : all_modules
     install_rpath : rpath_dirs,
     install_dir : modlibexecdir,
     dependencies : [thread_dep, libpulse_dep, libpulsecommon_dep, libpulsecore_dep, libintl_dep, platform_dep, platform_socket_dep] + extra_deps,
-    link_args : [nodelete_link_args, '-Wl,--no-undefined' ],
+    link_args : [nodelete_link_args],
     link_with : extra_libs,
     name_prefix : '',
     implicit_include_directories : false)
